cfme/tests/infrastructure/test_quota_tagging.py

```diff
@@ -317,3 +334,44 @@ def test_quota_infra(request, appliance, provider, setup_provider, admin_email,
     provision_request.wait_for_request(method='ui')
     request.addfinalizer(provision_request.remove_request)
     assert provision_request.row.reason.text == "Quota Exceeded"
+
+
+@pytest.mark.parametrize('context', [ViaSSUI, ViaUI])
+@pytest.mark.parametrize(
+    ['custom_prov_data'],
+    [
+        [{'hardware': {'memory': '4096'}}],
+        [{}],
```

Comment:
Question : 
What is the purpose of this empty array ?

Reply:
```[{}],``` - This parameterization is for selecting storage while provisioning VM/instance. But we don't need to select storage because by default storage is already more than assigned storage quota which is 2GB maximum.

Reply:
This logic needs to be in an in-line comment, there are a few assumptions in that statement that could breakdown as the application goes through changes.  Someone else trying to maintain this test won't be aware of the assumptions made.  Please either explicitly parametrize on storage of sufficient size, or include in-line comment that this parameter item is for storage, and the assumptions about default sizes and quota settings.

Reply:
@mshriver 
Thanks for your suggestion. It is really useful for someone new. I have added in-line comment for empty array as below.  Let me know if you have any suggestions around this comment.

> [{}],  # This parameterization is for selecting storage while provisioning VM.
        # But it is not possible to parameterize storage size.
        # Becuase it is defined in disk formats(Thin, Thick, Default and other types).
        # Also it varies with providers.
        # But we don't need to select storage because by default storage is already more than
        #  assigned storage quota which is 2GB maximum.